### DISCUSSION TITLE
Testfigures doesnt align

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#172] [Archived]
<USER>>>> snake<br/> <DATE> [March 29, 2020, 6:22pm]<br/> <br/> Hey, *slash* <br/> I've been running TTS on russian common voice dataset. *slash* <br/> Tensorbord shows that everything is looking good. However, TestFigures<br/> in IMAGE tab has no diagonal...<br/> <br/> *slash* <br/> <br/> *slash* <br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/testfigures-doesnt-align]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [March 29, 2020, 7:12pm]<br/> <br/> Did you give russian test sentences?<br/> <br/> check FAQ for using TTS in a new language *slash* <br/> <br/> ![](https://github.githubassets.com/favicon.ico){.site-icon width='32'<br/> <br/> ### [mozilla/TTS](https://github.com/mozilla/TTS/wiki/FAQ)<br/> <br/> forum: https://discourse.mozilla.org/c/tts) - mozilla/TTS<br/> <br/> BTW Isn't it a multi speaker dataset?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> snake<br/> <DATE> [March 30, 2020, 6:28am]<br/> <br/> test_sentences.txt has 4 lines with russian text. Shall I put it into<br/> root TTS folder or to output_path?<br/> <br/> it's multi speaker dataset, but multi speaker feature is disabled.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [March 30, 2020, 11:53am]<br/> <br/> you need to set the relevant field in the config file. Check the FAQ<br/> once more.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> snake<br/> <DATE> [March 31, 2020, 5:17pm]<br/> <br/> My config.json<br/> <br/> {<br/> 'model': 'Tacotron', // one of the model in models/ <br/> 'run_name': 'common_voice',<br/> 'run_description': 'tacotron cosntant stf parameters',<br/> <br/> // AUDIO PARAMETERS<br/> 'audio':{<br/> // Audio processing parameters<br/> 'num_mels': 80, // size of the mel spec frame.<br/> 'num_freq': 1025, // number of stft frequency levels. Size of the linear spectogram frame.<br/> 'sample_rate': 22050, // DATASET-RELATED: wav sample-rate. If different than the original data, it is resampled.<br/> 'win_length': 1024, // stft window length in ms.<br/> 'hop_length': 256, // stft window hop-lengh in ms.<br/> 'frame_length_ms': null, // stft window length in ms.If null, 'win_length' is used.<br/> 'frame_shift_ms': null, // stft window hop-lengh in ms. If null, 'hop_length' is used.<br/> 'preemphasis': 0.98, // pre-emphasis to reduce spec noise and make it more structured. If 0.0, no -pre-emphasis.<br/> 'min_level_db': -100, // normalization range<br/> 'ref_level_db': 20, // reference level db, theoretically 20db is the sound of air.<br/> 'power': 1.5, // value to sharpen wav signals after GL algorithm.<br/> 'griffin_lim_iters': 30,// #griffin-lim iterations. 30-60 is a good range. Larger the value, slower the generation.<br/> // Normalization parameters<br/> 'signal_norm': true, // normalize the spec values in range [0, 1]<br/> 'symmetric_norm': true, // move normalization to range [-1, 1]<br/> 'max_norm': 4.0, // scale normalization to range [-max_norm, max_norm] or [0, max_norm]<br/> 'clip_norm': true, // clip normalized values into the range.<br/> 'mel_fmin': 0.0, // minimum freq level for mel-spec. ~50 for male and ~95 for female voices. Tune for dataset!!<br/> 'mel_fmax': 8000.0, // maximum freq level for mel-spec. Tune for dataset!!<br/> 'do_trim_silence': true, // enable trimming of slience of audio as you load it. LJspeech (false), TWEB (false), Nancy (true)<br/> 'trim_db': 60 // threshold for timming silence. Set this according to your dataset.<br/> },<br/> <br/> // VOCABULARY PARAMETERS<br/> // if custom character set is not defined,<br/> // default set in symbols.py is used<br/> 'characters':{<br/> 'pad': '_',<br/> 'eos': '~',<br/> 'bos': '^',<br/> 'characters': 'АБВГДЕЖЗИЙКЛМНОПРСТУФХЦЧШЭЮЯабвгдежзийклмнопрстуфхцчшщъыьэюяёЁ',<br/> 'punctuations':'!'(),-.:;? ',<br/> 'phonemes':'iyɨʉɯuɪʏʊeøɘəɵɤoɛœɜɞʌɔæɐaɶɑɒᵻʘɓǀɗǃʄǂɠǁʛpbtdʈɖcɟkɡqɢʔɴŋɲɳnɱmʙrʀⱱɾɽɸβfvθðszʃʒʂʐçʝxɣχʁħʕhɦɬɮʋɹɻjɰlɭʎʟˈˌːˑʍwɥʜʢʡɕʑɺɧɚ˞ɫ'<br/> },<br/> <br/> // DISTRIBUTED TRAINING<br/> 'distributed':{<br/> 'backend': 'nccl',<br/> 'url': 'tcp: *slash* / *slash* /localhost:54321'<br/> },<br/> <br/> 'reinit_layers': [], // give a list of layer names to restore from the given checkpoint. If not defined, it reloads all heuristically matching layers.<br/> <br/> // TRAINING<br/> 'batch_size': 32, // Batch size for training. Lower values than 32 might cause hard to learn attention. It is overwritten by 'gradual_training'.<br/> 'eval_batch_size':16, <br/> 'r': 7, // Number of decoder frames to predict per iteration. Set the initial values if gradual training is enabled. <br/> 'gradual_training': [[0, 7, 32], [1, 5, 32], [50000, 3, 32], [130000, 2, 16], [290000, 1, 16]], // ONLY TACOTRON - set gradual training steps [first_step, r, batch_size]. If it is null, gradual training is disabled.<br/> 'loss_masking': true, // enable / disable loss masking against the sequence padding.<br/> <br/> // VALIDATION<br/> 'run_eval': true,<br/> 'test_delay_epochs': 10, //Until attention is aligned, testing only wastes computation time.<br/> 'test_sentences_file': 'test_sentences.txt', // set a file to load sentences to be used for testing. If it is null then we use default english sentences.<br/> <br/> // OPTIMIZER<br/> 'noam_schedule': false, // use noam warmup and lr schedule.<br/> 'grad_clip': 1.0, // upper limit for gradients for clipping.<br/> 'epochs': 1000, // total number of epochs to train.<br/> 'lr': 0.0002, // Initial learning rate. If Noam decay is active, maximum learning rate.<br/> 'wd': 0.000001, // Weight decay weight.<br/> 'warmup_steps': 4000, // Noam decay steps to increase the learning rate from 0 to 'lr'<br/> 'seq_len_norm': false, // Normalize eash sample loss with its length to alleviate imbalanced datasets. Use it if your dataset is small or has skewed distribution of sequence lengths.<br/> <br/> // TACOTRON PRENET<br/> 'memory_size': -1, // ONLY TACOTRON - size of the memory queue used fro storing last decoder predictions for auto-regression. If < 0, memory queue is disabled and decoder only uses the last prediction frame.<br/> 'prenet_type': 'original', // 'original' or 'bn'.<br/> 'prenet_dropout': true, // enable/disable dropout at prenet.<br/> <br/> // ATTENTION<br/> 'attention_type': 'original', // 'original' or 'graves'<br/> 'attention_heads': 4, // number of attention heads (only for 'graves')<br/> 'attention_norm': 'sigmoid', // softmax or sigmoid. Suggested to use softmax for Tacotron2 and sigmoid for Tacotron.<br/> 'windowing': false, // Enables attention windowing. Used only in eval mode.<br/> 'use_forward_attn': false, // if it uses forward attention. In general, it aligns faster.<br/> 'forward_attn_mask': false, // Additional masking forcing monotonicity only in eval mode.<br/> 'transition_agent': false, // enable/disable transition agent of forward attention.<br/> 'location_attn': true, // enable_disable location sensitive attention. It is enabled for TACOTRON by default.<br/> 'bidirectional_decoder': false, // use https://arxiv.org/abs/1907.09006. Use it, if attention does not work well with your dataset.<br/> <br/> // STOPNET<br/> 'stopnet': true, // Train stopnet predicting the end of synthesis.<br/> 'separate_stopnet': true, // Train stopnet seperately if 'stopnet==true'. It prevents stopnet loss to influence the rest of the model. It causes a better model, but it trains SLOWER.<br/> <br/> // TENSORBOARD and LOGGING<br/> 'print_step': 25, // Number of steps to log traning on console.<br/> 'save_step': 350, // Number of training steps expected to save traninpg stats and checkpoints.<br/> 'checkpoint': true, // If true, it saves checkpoints per 'save_step'<br/> 'tb_model_param_stats': false, // true, plots param stats per layer on tensorboard. Might be memory consuming, but good for debugging.<br/> <br/> // DATA LOADING<br/> 'text_cleaner': 'basic_cleaners',<br/> 'enable_eos_bos_chars': false, // enable/disable beginning of sentence and end of sentence chars.<br/> 'num_loader_workers': 8, // number of training data loader processes. Don't set it too big. 4-8 are good values.<br/> 'num_val_loader_workers': 8, // number of evaluation data loader processes.<br/> 'batch_group_size': 0, //Number of batches to shuffle after bucketing.<br/> 'min_seq_len': 0, // DATASET-RELATED: minimum text length to use in training<br/> 'max_seq_len': 150, // DATASET-RELATED: maximum text length<br/> <br/> // PATHS<br/> 'output_path': 'trainings/',<br/> <br/> // PHONEMES<br/> 'phoneme_cache_path': 'mozilla_ru_phonemes_2_1', // phoneme computation is slow, therefore, it caches results in the given folder.<br/> 'use_phonemes': false, // use phonemes instead of raw characters. It is suggested for better pronounciation.<br/> 'phoneme_language': 'ru', // depending on your target language, pick one from https://github.com/bootphon/phonemizer#languages<br/> <br/> // MULTI-SPEAKER and GST<br/> 'use_speaker_embedding': false, // use speaker embedding to enable multi-speaker learning.<br/> 'style_wav_for_test': null, // path to style wav file to be used in TacotronGST inference.<br/> 'use_gst': false, // TACOTRON ONLY: use global style tokens<br/> <br/> // DATASETS<br/> 'datasets': // List of datasets. They all merged and they get different speaker_ids.<br/> [<br/> {<br/> 'name': 'common_voice',<br/> 'path': '/audio',<br/> 'meta_file_train': 'validated.tsv',<br/> 'meta_file_val': 'test.tsv'<br/> }<br/> ]<br/> <br/> }<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> snake<br/> <DATE> [March 31, 2020, 5:19pm]<br/> <br/> `utils/text/symbols.py`<br/> <br/> ...<br/> _characters = 'АБВГДЕЖЗИЙКЛМНОПРСТУФХЦЧШЭЮЯабвгдежзийклмнопрстуфхцчшщъыьэюяёЁ! *slash* '(),-.:;? '<br/> ...<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> snake<br/> <DATE> [March 31, 2020, 5:36pm]<br/> <br/> Made *slash* ~300,000 steps. Model looks healthy, but when I run server.py and<br/> try to synthesize something I receive messed audio.<br/> <br/> *slash* <br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> snake<br/> <DATE> [April 1, 2020, 9:06am]<br/> <br/> I suspect that it might happen because of difference symbols of<br/> config.json and symbols.py<br/> <br/> config.json:<br/> <br/> 'characters': 'АБВГДЕЖЗИЙКЛМНОПРСТУФХЦЧШЭЮЯабвгдежзийклмнопрстуфхцчшщъыьэюяёЁ',<br/> <br/> symbols.py:<br/> <br/> _characters = 'АБВГДЕЖЗИЙКЛМНОПРСТУФХЦЧШЭЮЯабвгдежзийклмнопрстуфхцчшщъыьэюяёЁ! *slash* '(),-.:;? '<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> sanjaesc<br/> <DATE> [April 1, 2020, 9:25am]<br/> <br/> You can ignore the symbols.py if you are using characters in<br/> config.json. If you inspect symbols.py there is a make_symbols<br/> functions, which builds the symbols from the config.json. And since you<br/> have<br/> <br/> 'characters':'АБВГДЕЖЗИЙКЛМНОПРСТУФХЦЧШЭЮЯабвгдежзийклмнопрстуфхцчшщъыьэюяёЁ',<br/> 'punctuations':'!'(),-.:;? ',<br/> <br/> in your config.json, it should work just fine.<br/> <br/> I think the problem here is that you are using a multispeaker dataset<br/> but you have speaker embeddings disabled... see config<br/> 'use_speaker_embedding': false. *slash* <br/> Maybe you should try to train a single speaker dataset first and see if<br/> that works out for you.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> snake<br/> <DATE> [April 1, 2020, 9:34am]<br/> <br/> Can't ignore `symbols.py` because it's used during dataset loading. If I<br/> leave `symbols.py` with english chars it training won't start because<br/> input will be empty...<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> sanjaesc<br/> <DATE> [April 1, 2020, 9:48am]<br/> <br/> > in your config.json, it should work just fine.<br/> ><br/> > I think the problem here is that you are using a multispeaker dataset<br/> > but you have speaker embeddings disabled... see config<br/> > 'use_speaker_embedding': false. *slash* <br/> > Maybe you should try to train a single speaker dataset first and see<br/> > if that works out for you.<br/> <br/> I mean you dont need to modify symbols.py because symbols.py is using<br/> the characters from the config.json<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> snake<br/> <DATE> [April 1, 2020, 12:29pm]<br/> <br/> Ok. Cloned tts repo again. Started training again with minor changes in<br/> code. Haven't changed `symbols.py` file. We will see if it work out<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> snake<br/> <DATE> [April 2, 2020, 10:00am]<br/> <br/> Here is screenshots of new training.<br/> <br/> *slash* <br/> <br/> *slash* <br/> <br/> *slash* <br/> <br/> Is it ok? Shall I wait further?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> snake<br/> <DATE> [April 3, 2020, 5:35am]<br/> <br/> After 100,000 steps. *slash* <br/> <br/> *slash* <br/> <br/> As I understood the model trained good, but synthesize code might have<br/> issue to generate test snippets. I'll look thru this code<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> snake<br/> <DATE> [April 3, 2020, 5:41am]<br/> <br/> Logs during synthesizing<br/> <br/> tts_1 | | > Synthesizing test sentences<br/> tts_1 | | > Decoder stopped with 'max_decoder_steps<br/> tts_1 | | > Decoder stopped with 'max_decoder_steps<br/> tts_1 | | > Decoder stopped with 'max_decoder_steps<br/> tts_1 | | > Decoder stopped with 'max_decoder_steps<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> sanjaesc<br/> <DATE> [April 3, 2020, 7:55am]<br/> <br/> Are you still training on the common_voice dataset? I listened to some<br/> of the audio and would say the quality is not fit to train a tts system<br/> because of the noise in it.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [April 3, 2020, 8:25am]<br/> <br/> [rnnoise](https://github.com/xiph/rnnoise) removes a good amount of<br/> background noises (computer fan, electric hum, etc.).<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> snake<br/> <DATE> [April 3, 2020, 8:59am]<br/> <br/> Is there any other options for russian? I wouldn't say clips in common<br/> voice are noisy. At least in russian dataset...<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [April 3, 2020, 10:18am]<br/> <br/> training above looks broken. There might be a mismatch or a different<br/> problem in test time.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [April 3, 2020, 10:19am]<br/> <br/> have you tried rnnnoise for LJSpeech?<br/> <br/> [**[next page<br/> →](https://discourse.mozilla.org/t/testfigures-doesnt-align/56807?page=2)**<br/> <br/> [Archived Post]<br/>
